### PR TITLE
add runtime and service layer

### PR DESCRIPTION
**Pull request by ehmad-kamal-99**
implementing runtime and service layer

### CODE REVIEW

File: service/teacher.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+func (s *Service) deleteTeacher(id string) error {
+	err := s.rt.Db.DeleteTeacher(id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Service) updateTeacher(teacher *models.Teacher) error {
```
Comment (ahab94) [inline service/teacher.go]: All of these receivers should be exported.

File: service/student.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	return nil
+}
+
+func (s *Service) updateStudent(student *models.Student) error {
+	err := s.rt.Db.UpdateStudent(student)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```
Comment (ahab94) [inline service/student.go]: All of these receivers should be exported.

File: service/service_test.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Service{
+				rt: tt.fields.rt,
+			}
+			if err := s.updateTeacher(tt.args.teacher); (err != nil) != tt.wantErr {
+				t.Errorf("updateTeacher() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
```
Comment (ahab94) [inline service/service_test.go]: Add todos on each test that is empty for the moment.

File: service/service_test.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
+package service
```
Comment (ahab94) [inline service/service_test.go]: each student and teacher should have separate test class. `student_test` and `teacher_test`

File: service/service_test.go (hunk 3 above)
Comment (ehmad-kamal-99) [reply service/service_test.go]: But then linter says implement TODO, so I removed it for that reason.

File: service/service.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package service
+
+import runtime "github.com/ahab94/ApnaSchool"
+
+// Service initiates database service at runtime
+type Service struct {
+	rt *runtime.Runtime
```
Comment (ehmad-kamal-99) [inline service/service.go]: @ahab94 Should rt be exported type too?
